calabash-cucumber/lib/calabash-cucumber/store/preferences.rb

```diff
@@ -13,14 +14,80 @@ def initialize
         @path = File.join(dot_dir, "preferences", "preferences.json")
       end
 
+      # !@visibility private
+      def usage_tracking
+        preferences = read
+
+        unless valid_user_tracking_value?(preferences[:usage_tracking])
+          log_defaults_reset
+          preferences[:usage_tracking] = defaults[:usage_tracking]
+          write(preferences)
+        end
+
+        preferences[:usage_tracking]
+      end
+
+      # !@visibility private
+      def usage_tracking=(value)
+        if !valid_user_tracking_value?(value)
+          raise ArgumentError,
```

Comment:
I don't see how.  This will prevent callers from setting an invalid value.  There is no reason to reset defaults.  Maybe I don't understand the question?